### PR TITLE
[Refactor] (Pipeline) refactor the stream agg code

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/pipeline/exec/streaming_aggregation_sink_operator.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        std::shared_ptr<AggContext> agg_context)
+        : OperatorBuilder(id, name, exec_node),
+          _agg_node(exec_node),
+          _agg_context(std::move(agg_context)) {}
+
+OperatorPtr StreamingAggSinkOperatorBuilder::build_operator() {
+    return std::make_shared<StreamingAggSinkOperator>(this, _agg_node, _agg_context);
+}
+
+// use final aggregation source operator
```
Comment (qzsee) [inline be/src/pipeline/exec/streaming_aggregation_sink_operator.cpp:93]: fix these comment.

File: be/src/pipeline/exec/streaming_aggregation_source_operator.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+Status StreamingAggSourceOperator::close(RuntimeState* state) {
+    if (is_closed()) {
+        return Status::OK();
+    }
+    if (!_agg_node->decrease_ref()) {
+        _agg_node->release_resource(state);
+    }
+    return Operator::close(state);
```
Comment (qzsee) [inline be/src/pipeline/exec/streaming_aggregation_source_operator.cpp:71]: I think can call `Operator::close(state)` at the very beginning of a method

File: build.sh [hunk 3]
```diff
@@ -123,7 +123,7 @@ fi
 
 eval set -- "${OPTS}"
 
-PARALLEL="$(($(nproc) / 4 + 1))"
+PARALLEL="$(($(nproc) / 2 + 1))"
```
Comment (BiteTheDDDDt) [inline build.sh:126]: do not modify this

File: be/src/pipeline/exec/streaming_aggregation_source_operator.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/pipeline/exec/streaming_aggregation_source_operator.cpp:71]: just for return Status

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**